spec/02-integration/20-wasm/12-filters-as-plugins_spec.lua

```diff
@@ -0,0 +1,299 @@
+    assert.response(res).has.status(400)
+    return assert.response(res).has.jsonbody()
+  end
+
+  local function check_response(header_name, header_value, context)
+    helpers.wait_for_all_config_update()
+
+    local res, err = proxy:send({ path = "/status/200" })
+    assert(res and err == nil, tostring(err) .. ": " .. context)
+
+    assert.response(res).has.status(200)
+    local got = assert.response(res).has.header(header_name)
+    assert.equals(header_value, got, context)
+  end
+
+  local function check_filter_response(header_value, context)
+    return check_response(FILTER_HEADER, header_value, context)
+  end
+
+  local function admin_request(method, path, data)
+    local params = (data ~= nil and json_request(data)) or {}
+    params.method = method
+    params.path = path
+
+    local res, err = admin:send(params)
+    assert.is_nil(err)
+    return res
+  end
+
+  local function create_plugin(plugin)
+    local res = admin:post("/plugins", json_request(plugin))
+    assert.response(res).has.status(201)
+    return assert.response(res).has.jsonbody()
+  end
+
+  local function update_plugin(id, plugin)
+    local res = admin:patch("/plugins/" .. id, json_request(plugin))
+    assert.response(res).has.status(200)
+    return assert.response(res).has.jsonbody()
+  end
+
+  local function get_plugin(endpoint, id)
+    local res = admin:get("/plugins/" .. id)
+    assert.response(res).has.status(200)
+    return assert.response(res).has.jsonbody()
+  end
+
+  local function create_filter_chain(fc)
+    local res = admin:post("/filter-chains", json_request(fc))
+    assert.response(res).has.status(201)
+    return assert.response(res).has.jsonbody()
+  end
+
+
+  lazy_setup(function()
+    require("kong.runloop.wasm").enable({
+      { name = "response_transformer",
+        path = FILTER_PATH .. "/response_transformer.wasm",
+      },
+    })
+
+    bp, db = helpers.get_db_utils(strategy, {
+      "routes",
+      "services",
+      "filter_chains",
+      "plugins",
+    })
+
+    helpers.start_kong({
+      database = strategy,
+      nginx_conf = "spec/fixtures/custom_nginx.template",
+      nginx_main_worker_processes = "2",
+      wasm = true,
+      wasm_filters = "response_transformer",
+      plugins = "response-transformer",
+    })
+
+    admin = helpers.admin_client()
+    proxy = helpers.proxy_client()
+    proxy.reopen = true
+  end)
+
+  lazy_teardown(function()
+    if admin then
+      admin:close()
+    end
+
+    if proxy then
+      proxy:close()
+    end
+
+    helpers.stop_kong()
+  end)
+
+  before_each(function()
+    -- create a scratch service/route to use
+    service = assert(bp.services:insert({}))
+    route = assert(bp.routes:insert({
+      paths = { "/" },
+      service = service,
+    }))
+  end)
+
+  after_each(function()
+    bp.filter_chains:truncate()
+    bp.plugins:truncate()
+
+    db.routes:delete(route)
+    db.services:delete(service)
+  end)
+
+  describe("/plugins", function()
```

Comment:
What about a test to get all plugins and include 1 or more wasm filters and 1 or more lua plugins?

Reply:
Added a new test in 7c13e4461651d49212cc00754cbf9fb3fd0b11a2. Is this what you were looking for?

Reply:
Yep, thanks! LGTM!